front/src/services/api/hallPlanFetch.js

```diff
@@ -319,52 +319,191 @@ export const fetchHallPlan = (seanceInfo) =>
         })
 
 export const addCinema = (cinemaInfo) =>
-    delay(500)
-        .then(() => {
-            let existedCity = cinemaHallPlans.find(city => city.cityName === cinemaInfo.city);
-            if (!existedCity){
-                existedCity = {
-                    cityName: cinemaInfo.city,
-                    cinemas: [
+    fetch(
+        "./api/cities",
+        {
+            method: 'post',
+            headers: {
```

Comment:
you setting this up almost every request. isn't it better to optimize it somehow?